org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/part/WindowContainer.java

```diff
@@ -635,6 +639,38 @@ public Tab insertAfter(Tab refTab, Component comp, String title, boolean closeab
     		return insertBefore((Tab)refTab.getNextSibling(), comp, title, closeable, enable, decorateInfo);
     }
 
+    /**
+     * IDEMPIERE-5275 - Tabular Report Re-Run button/close parameter window
+     * @param refTab 
+     * @param comp
+     * @param title
+     * @return
+     */
+    public Tab replace(Tab refTab, Window comp, String title) {
+    	 
+         if (refTab == null)  
+         {
+         	tabbox.getTabs().getChildren();
+         }
+         else
+         {
+         	org.zkoss.zul.Tabpanel refpanel = refTab.getLinkedPanel();
+         	Component oldWindow = refpanel.getFirstChild();
+     		if(oldWindow instanceof ProcessDialog)
+     			((ProcessDialog)oldWindow).unlockUI(null);
+     		else if(oldWindow instanceof ZkReportViewer)
+				((ZkReportViewer)oldWindow).hideBusyMask();
+			else if(oldWindow instanceof AbstractADWindowContent)
+				((AbstractADWindowContent)oldWindow).hideBusyMask();
+     		((Window) oldWindow).onClose();
+     		comp.setParent(refpanel);
```

Comment:
the change make for this block is wrong. The previous if (oldWindow instanceof Window) condition is correct, what I mean is you should have the else part too, i.e when oldWindow is not instanceof Window, you should just detach oldWindow (probably sounds more readable if that's rename to firstChild instead) and attach comp to refpanel.